.github/workflows/push-mimir-build-image.yml

```
@@ -0,0 +1,103 @@
+name: Build and Push mimir-build-image
+
+# configure trigger by pull request or manually
+on:
+  pull_request:
+    types: [opened, synchronize, ready_for_review]
+    paths:
+      - mimir-build-image/Dockerfile  
+
+jobs: 
+  build_and_push:
+    runs-on: ubuntu-latest
+    # If the PR is raised by dependabot, then build the image
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v3
+
+      - name: Checkout Pull Request Branch
+        run: gh pr checkout ${{ github.event.pull_request.number }}
+        env:
+          GITHUB_TOKEN: ${{secrets.GITHUB_TOKEN}}
+
+      - name: Setup QEMU
+        uses: docker/setup-qemu-action@v2
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+
+      - name: Login to Docker Hub
+        uses: docker/login-action@v2
+        with:
+          username: ${{ secrets.DOCKER_USERNAME }}
+          password: ${{ secrets.DOCKER_PASSWORD }}
+
+      - name: Prepare Variables
+        id: prepare
+        run: |
+          echo "path=mimir-build-image/Dockerfile" >> $GITHUB_OUTPUT
+          echo "image=grafana/mimir-build-image" >> $GITHUB_OUTPUT
+
+      - name: Compute Image Tag
+        id: compute_hash
+        run: |
+          current_hash=$(md5sum ${{ steps.prepare.outputs.path }} | awk '{print $1}')
+          echo "the file path is ${{ steps.prepare.outputs.path }}"
+          echo "build tag is $current_hash"
+          tag="${{ github.event.pull_request.number }}_$current_hash"
+          echo "tag=$tag" >> $GITHUB_OUTPUT
+
+      - name: Check Should Build Image
+        id: check_build
+        run: |
+          echo "Checking if image should be built"
+          image="docker.io/${{ steps.prepare.outputs.image }}:${{ steps.compute_hash.outputs.tag }}"
+          if skopeo inspect --raw "docker://$image" >/dev/null 2>&1; then
+            echo "build=false" >> $GITHUB_OUTPUT
+            echo "Tag exists"
+          else
+            echo "Tag does not exist"
+            echo "build=true" >> $GITHUB_OUTPUT
+          fi
+
+      - name: Add Comment to the PR
+        id: notification
+        run: | 
+          if [[${{ steps.check_build.outputs.build == 'true' }}]]; then
+            gh pr comment $PR_URL --body "**This PR triggers the build of the mimir-build-image**
+            image would be auto built and pushed to docker hub, a commit would be added to PR with new image version."
```

I would modify the message to active message: "Building new version of mimir-build-image. Image will be pushed to docker hub and commit will be added to the PR with new image version".

(I would also not mention that "This PR" triggers something... as more commits to the PR can also trigger that.)

```suggestion
            gh pr comment $PR_URL --body "**Building new version of mimir-build image**. After image is built and pushed to docker hub, new commit will be added to this PR with new image version <insert image tag here>."
```